src/cast_core/src/ExternalScriptRunner.cpp

```diff
@@ -66,29 +67,72 @@ struct ExternalScriptRunner::PhaseConfig {
 std::string ExternalScriptRunner::exec(const char* cmd) {
     std::array<char, 128> buffer;
     std::string result;
-    std::unique_ptr<FILE, decltype(&pclose)> pipe(popen(cmd, "r"), pclose);
+
+    FILE* pipe = popen(cmd, "r");
     if (!pipe) {
         throw std::runtime_error("Execution of command " + std::string(cmd) + " failed!");
     }
-    while (fgets(buffer.data(), buffer.size(), pipe.get()) != nullptr) {
-        result += buffer.data();
+    try {
+        while (fgets(buffer.data(), buffer.size(), pipe) != nullptr) {
+            result += buffer.data();
+            BOOST_LOG_TRIVIAL(info) << "Script output: " << buffer.data();
+        }
     }
+    catch (...) {
+        pclose(pipe);
+        throw;
+    }
```

Comment:
Would it be worth creating a RAII wrapper for this pipe?

Reply:
It's a good thought. I think in this case, since I need the results from pclose to get the exitStatus, I'd need to track the state of whether or not the pipe is closed in the class and still call some sort of close method manually. My vote would be to keep the current implementation since it's relatively contained. Let me know if you disagree and I'm happy to add the new class.

Reply:
Fair, sounds good. Can always add one later if it turns out to be needed